services/src/main/java/org/keycloak/keys/JavaKeystoreKeyProviderFactory.java

```diff
@@ -159,15 +160,29 @@ private static void validateCertificateChain(List<X509Certificate> certificates)
 
     // merge the algorithms supported for RSA and EC keys and provide them as one configuration property
     private static ProviderConfigProperty mergedAlgorithmProperties() {
-        List<String> algorithms = Stream.of(
-                        List.of(Algorithm.AES, Algorithm.EdDSA),
-                        List.of(Algorithm.ES256, Algorithm.ES384, Algorithm.ES512),
-                        Attributes.HS_ALGORITHM_PROPERTY.getOptions(),
-                        Attributes.RS_ALGORITHM_PROPERTY.getOptions(),
-                        Attributes.RS_ENC_ALGORITHM_PROPERTY.getOptions(),
-                        GeneratedEcdhKeyProviderFactory.ECDH_ALGORITHM_PROPERTY.getOptions())
-                .flatMap(Collection::stream)
-                .toList();
+        List<String> algorithms;
```

Comment:
Don't duplicate current lists of lists, but rather add a condition to append to the original list of Feature.PQC_MLDSA feature is enabled. Using `Stream.of` or just creating a new linked list and adding current + ML-DSA to it.